engine/src/main/protobuf/NetMessage.proto

```diff
@@ -146,6 +146,8 @@ message ServerInfoMessage {
     repeated WorldInfo worldInfo = 9;
     repeated sint32 biomeShortId = 10 [packed=true];
     repeated string biomeId = 11;
+    repeated string blackList = 12;
```

Comment:
Not IpAddress but clientId.   So "blacklistedClientId" and "whitelistedClientId".

Although I'm not certain what the point of blacklisting a client id would be since it'd be trivial to generate a new one.

Reply:
Now that I've looked at all the code, I am wondering why we are passing the whitelist and blacklist to the client.   If it's just to indicate what the error message states, I recommend sending an error message instead.   It's just asking for trouble to send the list of whitelisted client ids to a client who was rejected.   A clever person would just copy a whitelisted client id into their client.

Reply:
Now that I think about it, you are right. I should probably be sending an error message across instead.